libraries/ms-common/inc/soft_timer.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <stdint.h>
+
+#include "status.h"
+
+#define SOFT_TIMER_NUM 10
+
+typedef void (*soft_timer_callback)(uint8_t timer_id, void* context);
+
+// Initializes a set of software timers. Clock speed should be in MHz. Subsequent calls will do
+// nothing. The clock speed is that of the external PLL crystal.
+void soft_timer_init(uint8_t clock_speed);
```

Comment:
No need to abstract there is already just a function for it.

No reason to check, I'll remove that initialization stuff. I will add a function to check if there are timer still running as well. I just didn't want people to reset when timers were still running.